src/Framework/NativeMethods.cs

```diff
@@ -594,17 +594,19 @@ private static void SetMaxPath()
         }
     }
 
-    internal static bool IsMaxPathLegacyWindows()
+    internal static bool TryIsLongPathsEnabledRegistry()
```

Comment:
In .NET, the `Try` prefix is a convention for a method whose return value is a bool that indicates success or failure, for example

```c#
class Int32
{
    bool TryParse(string? s, out int result)
}
```

The expectation is that the caller checks the return value to learn whether the out param has a good value in it or not.

So I'd rather avoid `Try` prefixes in cases that aren't like that.

Reply:
Actually I'm not sure I understand this refactor in general. In `Xmake.cs` you're already wrapping the logging in an `IsWindows` call, could you use the result of `IsLongPathsEnabledRegistry()` instead?

Reply:
The point was that I wanted 2 things:
1. If we're on Windows display the value of the registry.
2. If we're somewhere else don't display anything.
I refactored it with an enum to clarify and removed the Try.